models/remoteworkitem.go

```diff
@@ -0,0 +1,21 @@
+package models
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}	
+}
+
+// The intregration code for a specific remote provider must populate this.
+type WorkItemMap map[string]interface{}
+
+// Map maps RemoteWorkItem to WorkItem
+func (rwi RemoteWorkItem) MapRemote (wiMap WorkItemMap, wiType WorkItemType) WorkItem {
+	workItem := WorkItem{
+		Type: wiType.ID
+	}	
+	
+	for from_key, to_key := range wiMap {
+		workItem.Fields[to_key] = remoteWorkItem.Fields[from_key]
```

Comment:
Check with @DhritiShikhar @Ritsyy , but from_key might need to be resolved as a child element of something. Don't remember how the github/jira issues look but as an example

```json
issue: {
  meta: {
    title: ""
  }
}
```

The map might need to be able to express: "issue/meta/title" -> "system.title". Depending on how the GitHub/Jira/Trello/Bugzilla structure is for the 3 fields of interested. If they are all Flat 1. level then we don't need it yet, if they are not we do.. :)